indexing-service/src/main/java/org/apache/druid/indexing/overlord/ImmutableWorkerInfo.java

```diff
@@ -119,12 +142,33 @@ public boolean isValidVersion(String minVersion)
     return worker.getVersion().compareTo(minVersion) >= 0;
   }
 
-  public boolean canRunTask(Task task)
+  public boolean canRunTask(Task task, double parallelIndexWorkerRatio)
   {
     return (worker.getCapacity() - getCurrCapacityUsed() >= task.getTaskResource().getRequiredCapacity()
+            && canRunParallelIndexTask(task, parallelIndexWorkerRatio)
             && !getAvailabilityGroups().contains(task.getTaskResource().getAvailabilityGroup()));
   }
 
+  public boolean canRunParallelIndexTask(Task task, double parallelIndexWorkerRatio)
+  {
+    if (!task.getType().equals(ParallelIndexSupervisorTask.TYPE)) {
+      return true;
+    }
+    return getWorkerParallelIndexCapacity(parallelIndexWorkerRatio) - getCurrParallelIndexCapacityUsed()
+           >= task.getTaskResource().getRequiredCapacity();
+
+  }
+
+  private int getWorkerParallelIndexCapacity(double parallelIndexWorkerRatio)
+  {
+    int totalCapcity = worker.getCapacity();
+    int workerParallelIndexCapacity = (int) Math.round(parallelIndexWorkerRatio * totalCapcity);
+    if (totalCapcity > 1 && workerParallelIndexCapacity == 0) {
```

Comment:
```suggestion
    if (workerParallelIndexCapacity < 1) {
```

Number of task slots for parallel indexing task should be atleast 1, even if totalCapacity is 1.

Reply:
Nit: totalCapcity to totalCapacity